addons/sourcemod/scripting/vsh/bosses/boss_samyro.sp

```diff
@@ -0,0 +1,219 @@
+#define SAMYRO_MUSIC	"vsh_rewrite/samyro/samyro_music.mp3"
+
+static int g_iSamyroModelHat;
+static int g_iSamyroModelMask;
+static int g_iSamyroModelKatana;
+static int g_iSamyroModelHands;
+
+static float g_flClientRageGainLastTime[TF_MAXPLAYERS + 1];
+
+static char g_strSamyroRoundStart[][] =  {
+	"vo/pyro_battlecry01.mp3", 
+	"vo/pyro_battlecry02.mp3"
+};
+
+static char g_strSamyroRoundWin[][] =  {
+	"vo/taunts/pyro/pyro_taunt_head_pain_21.mp3", 
+	"vo/taunts/pyro/pyro_taunt_head_pain_22.mp3"
+};
+
+static char g_strSamyroRoundLose[][] =  {
+	"vo/pyro_paincrticialdeath01.mp3", 
+	"vo/pyro_paincrticialdeath02.mp3", 
+	"vo/pyro_paincrticialdeath03.mp3", 
+	"vo/taunts/pyro/pyro_taunt_rps_lose_03.mp3"
+};
+
+static char g_strSamyroRage[][] =  {
+	"vo/pyro_laughlong01.mp3"
+};
+
+static char g_strSamyroKill[][] =  {
+	"vo/taunts/pyro_taunts01.mp3", 
+	"vo/taunts/pyro_taunts02.mp3", 
+	"vo/taunts/pyro_taunts03.mp3", 
+	"vo/taunts/pyro_taunts04.mp3", 
+	"vo/compmode/cm_pyro_pregamelostlast_02.mp3", 
+	"vo/compmode/cm_pyro_pregamelostlast_03.mp3"
+};
+
+static char g_strSamyroLastMan[][] =  {
+	"vo/cm_pyro_pregamewonlast_01.mp3"
+};
+
+static char g_strSamyroAbility[][] =  {
+	"items/samurai/tf_samurai_noisemaker_seta_01.wav", 
+	"items/samurai/tf_samurai_noisemaker_seta_02.wav", 
+	"items/samurai/tf_samurai_noisemaker_seta_03.wav"
+};
+
+methodmap CSamyro < SaxtonHaleBase
+{
+	public CSamyro(CSamyro boss)
+	{
+		boss.CallFunction("CreateAbility", "CWallClimb");
+		
+		CAddCond abilityCond = boss.CallFunction("CreateAbility", "CAddCond");
+		abilityCond.AddCond(TFCond_RuneAgility);
+		abilityCond.bRemoveOnRage = true;
+		
+		CRageAddCond rageCond = boss.CallFunction("CreateAbility", "CRageAddCond");
+		rageCond.flRageCondDuration = 12.0;
+		rageCond.flRageCondSuperRageMultiplier = 14.0 / 12.0;
+		rageCond.AddCond(TFCond_RuneHaste);
+		rageCond.AddCond(TFCond_DefenseBuffed);
+		
+		boss.iBaseHealth = 700;
+		boss.iHealthPerPlayer = 650;
+		boss.nClass = TFClass_Pyro;
+		boss.iMaxRageDamage = 2500;
+		
+		AddCommandListener(Command_DropItem, "dropitem");
+	}
+	
+	public void GetBossName(char[] sName, int length)
+	{
+		strcopy(sName, length, "Samyro");
+	}
+	
+	public void GetBossInfo(char[] sInfo, int length)
+	{
+		StrCat(sInfo, length, "\nHealth: Low");
+		StrCat(sInfo, length, "\n ");
+		StrCat(sInfo, length, "\nAbilities");
+		StrCat(sInfo, length, "\n- Passive Rage Gain");
+		StrCat(sInfo, length, "\n- Wall Climb");
+		StrCat(sInfo, length, "\n- Agility");
```

Comment:
Doesnt really tell much about this, "Agility powerup on alt attack" or similar should be enough